### PR TITLE
Team a login page business logic

### PR DESCRIPTION
**Pull request by alphaLynx97**
image
image
image
This implements the use of a login token for the moodle server using the old intelligrade logic. So that means bringing in their moodle api singleton and other necessary files and editing to match our software.

### CODE REVIEW

File: teamA/lib/Views/login_page.dart [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+              ),
+            ),
+            const SizedBox(height: 32),
+
+            // Login Button
+            ElevatedButton(
+              onPressed: () async {
+                var wasSuccessful = await LoginScreen.controller.loginToMoodle(_usernameController.text, _passwordController.text);
+                if (wasSuccessful) {
+                Navigator.push(
```
Comment (DavidWorthy) [inline teamA/lib/Views/login_page.dart]: Might want to also pull the course list here like TestWizard did. I believe we talked about that during the meeting on Wednesday.

**Review by DavidWorthy — CHANGES_REQUESTED** [review]
As far as I can tell, the login works. However, there's a part missing to the login process, namely obtaining the list of the user's courses like TestWizard did on login.

File: teamA/lib/Views/login_page.dart [hunk 2]
```diff
@@ -0,0 +1,189 @@
+import 'package:flutter/material.dart';
```
Comment (AlmostCarbon-14) [inline teamA/lib/Views/login_page.dart:1]: Are these Anthony's changes that've not yet been merged?

**Comment by DavidWorthy** [comment]
Fixed the inconsistency in pubspec.yaml so that merging will be easier.

**Comment by scottmcglynn** [comment]
This branch will not compile in MacOS mode on my machine. I get numerous CocoaPods errors but the core TeamA branch compiles fine. I'd like to fix that if we can figure it out before merging.

File: teamA/lib/Views/login_page.dart (hunk 1 above)
Comment (alphaLynx97) [reply teamA/lib/Views/login_page.dart]: image
Will this suffice for now? Stored as a local variable but the controller can be called to retrieve the courses at any time after login. The usertoken is stored in the singleton

Comment (alphaLynx97) [reply teamA/lib/Views/login_page.dart]: image

File: teamA/lib/Views/login_page.dart (hunk 2 above)
Comment (alphaLynx97) [reply teamA/lib/Views/login_page.dart:1]: Yea I just reached into his branch and integrated it on mine.

**Comment by scottmcglynn** [comment]
Confirmed; build in MacOS works now.

**Review by AlmostCarbon-14 — APPROVED** [review]

**Review by DavidWorthy — APPROVED** [review]
Excellent. I should be able to use the main controller for my own page's business logic.

**Pull request merged**